tests/services/profile.service.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { Profile } from '../../src/app/core/models/profile.model';
+import { GithubService } from '../../src/app/core/services/github.service';
+import { ProfileService } from '../../src/app/core/services/profile.service';
+
+let githubService: GithubService;
+let profileService: ProfileService;
+
+const EMPTY_PROFILES = [];
+const VALID_PROFILES = [
+  <Profile>{
+    profileName: 'CATcher',
+    encodedText: 'CATcher-org/public_data'
+  }
+];
+const INVALID_PROFILES = [
+  {
+    profileName: 'CATcher'
+  }
+];
+
+describe('ProfileService', () => {
+  beforeEach(() => {
+    githubService = new GithubService(null, null, null);
+    profileService = new ProfileService(githubService);
+  });
+
+  describe('.fetchExternalProfiles()', () => {
+    it('should return the set of profiles if no profiles are given', () => {
+      spyOn(window, 'fetch').and.returnValue(generateProfilesPromise(EMPTY_PROFILES));
+      return profileService.fetchExternalProfiles().then((profiles) => expect(profiles).toEqual(EMPTY_PROFILES));
+    });
+
+    it('should return the set of profiles if a set of valid profiles are given', () => {
+      spyOn(window, 'fetch').and.returnValue(generateProfilesPromise(VALID_PROFILES));
+      return profileService.fetchExternalProfiles().then((profiles) => expect(profiles).toEqual(VALID_PROFILES));
+    });
+
+    it('should return the set of profiles if a set of valid profiles are given', () => {
+      spyOn(window, 'fetch').and.returnValue(generateProfilesPromise(INVALID_PROFILES));
+      return profileService.fetchExternalProfiles().catch((error: Error) => expect(error.message).toEqual('profiles.json is malformed'));
```

Comment:
This test won't fail if an error is *not thrown*.
We can add `.then(() => fail())`
[Reference to Jasmine docs](https://jasmine.github.io/api/3.7/global.html#fail)

Reply:
Alright! :+1: